frontend/views/containers/dashboard/GroupMembers.vue

```diff
@@ -54,45 +42,64 @@
 import { mapGetters } from 'vuex'
 import { OPEN_MODAL } from '@utils/events.js'
 import sbp from '~/shared/sbp.js'
+import Avatar from '@components/Avatar.vue'
 import AvatarUser from '@components/AvatarUser.vue'
-import GroupMemberMenu from '@containers/dashboard/GroupMemberMenu.vue'
-import Tooltip from '@components/Tooltip.vue'
+import GroupMembersMenu from '@containers/dashboard/GroupMembersMenu.vue'
+import GroupMembersTooltipPending from '@containers/dashboard/GroupMembersTooltipPending.vue'
 
 export default {
   name: 'GroupMembers',
   components: {
+    Avatar,
     AvatarUser,
-    GroupMemberMenu,
-    Tooltip
-  },
-  methods: {
-    invite () {
-      this.$router.push({ path: '/invite' })
-    },
-    openModal (modal, queries) {
-      sbp('okTurtles.events/emit', OPEN_MODAL, modal, queries)
-    }
+    GroupMembersMenu,
+    GroupMembersTooltipPending
   },
   computed: {
     ...mapGetters([
       'groupProfiles',
       'groupShouldPropose',
       'groupMembersCount',
-      'ourUsername'
+      'ourUsername',
+      'userDisplayName',
+      'currentGroupState',
+      'groupMembersPending'
     ]),
+    weJoinedMs () {
+      return new Date(this.currentGroupState.profiles[this.ourUsername].joinedDate).getTime()
+    },
     firstTenMembers () {
       const profiles = this.groupProfiles
-      const usernames = Object.keys(profiles).slice(0, 10)
-      return usernames.reduce((acc, username) => {
+      const sliceIndex = 10 - Math.min(10, Object.keys(this.groupMembersPending).length) // avoid slicing too many members.
+      const usernames = Object.keys(profiles).slice(0, sliceIndex)
```

Comment:
Is this change necessary? AFAICT `.slice` will automatically bounds check for you, you don't need to do it yourself.

Reply:
No... Let's imagine we have an array `[1,2,3,4,5]` and we do `slice(0, -3)`, the output will be `[1, 2]`. IRL, this can happen if we have more than 10 members pending. Run the code on your computer and see it by yourself.

Reply:
Oh, I see what you mean. Oops, misread the code. Nice job! 👍